### ISSUE TITLE
Tile movement

### ISSUE THREAD

**Issue by Zumorica** [space-wizards/space-station-14#5551]
## Description
We should implement tile-based movement, and add a reagent that makes you have it.

**Screenshots**
![image](https://user-images.githubusercontent.com/6766154/143592175-1fca9707-c9c2-465e-8302-56f5a3e1c333.png)
![image](https://user-images.githubusercontent.com/6766154/143592211-f4605912-fd39-4969-a95b-bbbe8fe8579f.png)
![image](https://user-images.githubusercontent.com/6766154/143592226-63de98ba-ebec-4891-8c6f-f8e8a1fb5311.png)

**Additional context**
when the

**Comment by ike709** [comment]
Make it a cvar for the servers who want tile movement and then the reagent can give you pixel movement.

**Comment by Zumorica** [comment]
Here's an old proof of concept I coded in less than half an hour, btw.

![Peek_13-01-2022_17-07](https://user-images.githubusercontent.com/6766154/161649207-4fd72c33-b6cc-4395-a266-6ad4c6f8fd18.gif)

**Comment by Zumorica** [comment]
As discussed in the June 11th, 2022 maintainer meeting, we should, unironically, implement tile movement.

- Keep doing distance measurements for interactions.
    - Have simpler physics for it, "hard" vs "not hard" without proper shapes like we have right now.
        - Have support for things such as windoors, etc. (This is not trivial)
- Have a simple, working implementation, forks can improve it further if they want.
    - We will upstream/merge good changes to it, as long as the code doesn't get diverged *too much*.
        - Making code more modular, using events and entity systems would still help here.

**Comment by ShiftyRail** [comment]
It's very encouraging to learn that SS14 will have tile-based movement. This makes it a lot easier to convince people to move on to the new game. Good to hear. 👍

**Comment by moonheart08** [comment]
> It's very encouraging to learn that SS14 will have tile-based movement. This makes it a lot easier to convince people to move on to the new game. Good to hear. 👍

To clarify, it'll support it, but we're not going to enable it on our servers, except maybe on a demo server.

**Comment by ShiftyRail** [comment]
Yes, but for eventual forks and new contributors this is still very good news.
